### PR TITLE
Register missing encoding providers for core host

### PR DESCRIPTION
**Pull request by genlu**
Fix https://github.com/dotnet/roslyn/issues/54985

### CODE REVIEW

**Review by tmat — APPROVED** [review]

File: src/Workspaces/Remote/ServiceHub/Services/Host/RemoteHostService.cs [hunk 1]
```diff
@@ -64,6 +64,9 @@ static RemoteHostService()
             // we set up logger here
             RoslynLogger.SetLogger(new EtwLogger(s_logChecker));
 
+            // Make encodings that is by default present in desktop framework but not in corefx available to runtime.
```
Comment (kasperk81) [inline src/Workspaces/Remote/ServiceHub/Services/Host/RemoteHostService.cs:67]:
```suggestion
            // Make encodings that is by default present in desktop framework but not in corehost available to runtime.
```

**Comment by kasperk81** [comment]
why did you ignore the typo comment @genlu?

**Comment by genlu** [comment]
@kasperk81 It's not a typo: the encodings are missing not because we are running on servicehub core host, it's actually the default in corefx.

**Comment by kasperk81** [comment]
Ok, though these days corefx is called runtime libraries.

**Pull request merged**